src/main/java/com/clairvoyant/weather/handler/WeatherHandler.java

```diff
@@ -0,0 +1,79 @@
+package com.clairvoyant.weather.handler;
+
+import com.clairvoyant.weather.dto.WeatherDto;
+import com.clairvoyant.weather.service.WeatherService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.server.ServerRequest;
+import org.springframework.web.reactive.function.server.ServerResponse;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+/**
+ * @author Gufran Khan
+ * @version 1.0
+ * @date 23-08-2021 12:26
+ */
+@Component
+@Slf4j
+public class WeatherHandler {
+
+  private final WeatherService weatherService;
+
+  @Autowired
+  public WeatherHandler(WeatherService weatherService) {
+    super();
+    this.weatherService = weatherService;
+  }
+
+  public Mono<ServerResponse> getAllWeatherDetails(ServerRequest serverRequest) {
```

Comment:
this is applicable to all the methods. Please add comments on the method describing what is it doing.